dev-docs/hail-for-new-engineers.md

```diff
@@ -0,0 +1,189 @@
+introduced. During this time, Hail was not versioned. Users had to build and use Hail directly from
+the source code repository.
+
+In March of 2017, Hail team began work on a compiler.
+
+In October of 2018, the Hail Python interface was modified, in a backwards-incompatible way. This
+new Python interface was named "Hail 0.2". The old Python interface was retroactively named "Hail
+0.1". Hail team began deploying a Python package named `hail` to [PyPI](https://pypi.org). The Hail
+python package version compiles with [Semantic Versioning](https://semver.org).
+
+Meanwhile, in September of 2018, Hail team began work on a system called "Batch". Batch runs
+programs in parallel on a cluster of virtual machines. Also in September, Hail team began work on a
+system called "CI". CI is an acronym for Continuous Integration. CI runs the tests for every pull
+request (PR) into the `main` branch of [`hail-is/hail`](https://github.com/hail-is/hail). CI
+automatically merges into main any pull request that both passes the tests and has at least one
+"approving" review and no "changes requested" reviews. CI uses Hail Batch to run the tests.
+
+Around this time, the Hail team split organized itself into two sub-teams: "compilers" team and
+"services" team. The compilers team is responsible for the Hail Python library, the compiler, and
+the associated runtime. The compilers team code is entirely contained in the `hail` directory of the
+`hail-is/hail` repository. The services team is responsible for Batch, CI, and the software
+infrastructure that supports those services. Each service has its own directory in the hail
+repository. More information about the structure of the repository can be found in
+[`hail-overview.md`](hail-overview.md).
+
+Beginning in early 2020, beta users were given access to Hail Batch.
+
+In April of 2020, the Hail team began referring to the Hail python library as "Hail Query" and
+created a system called "the Query Service". Inside the Hail Query Service, Hail Query uses Hail
+Batch to run an analysis across many computer cores instead of using Apache Spark.
+
+## Project Goals
+
+Hail exists to accelerate research on the genetics of human disease. We originally focused on the
+needs of statistical geneticists working with very large datasets of human genetic sequences. This
+project lead to the development of Hail Query.
+
+## Hail Products, Briefly
+
+Hail team maintains two software systems which our users directly use: Query and Batch
+
+### Hail Query
+
+Hail Query is a Python library for the analysis of large datasets. In Hail Query, a dataset is
+represented as Table or a Matrix Table.
+
+Hail Tables are similar to SQL tables, Pandas Dataframes, Excel spreadsheets, and CSV files.
+
+Hail Matrix Tables do not have analogues in most other systems. Perhaps the only analogue is a
+[SciDB](https://dbdb.io/db/scidb) multi-dimensional array. A Hail Matrix Table can represent dense,
```

Comment:
Is SciDB the same as TileDB? I think they have something analogous.

Reply:
The history is confusing and not entirely clear to me. I believe Stonebreaker (of SciDB fame) worked with Papadopoulos at one point. I believe TileDB is at the least inspired by SciDB. TileDB, unlike SciDB, is a file format with a library to read and write it. There is also GenomicsDB which is built on top of or forks TileDB. I'll add some info.